src/lib/drawer/DrawerProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useContext, useReducer } from 'react';
+import { drawerActions, DrawerActions, DrawerReducer } from './Drawer';
```

Comment:
Seems like we have some circular imports here. While it may not cause any problems with the build, maybe it would make since to create separate files for DrawerReducer and Drawer.types?